CMakeLists.txt

```diff
@@ -316,7 +316,12 @@ if(WIN32 AND (MINGW OR CYGWIN))
 endif()
 
 if(BUILD_FOR_R)
+  if(MSVC)
+    # https://docs.microsoft.com/en-us/cpp/build/reference/link-input-files?redirectedfrom=MSDN&view=vs-2019
+    TARGET_LINK_LIBRARIES(_lightgbm ${CMAKE_CURRENT_BINARY_DIR}/R.lib)
```

Comment:
I believe we should return this path from `FindLibR.cmake` module (guess better in a separate variable than overwriting `LIBR_CORE_LIBRARY`) because it is hidden deep in `create_rlib_for_msvc` function and outer environment should know nothing about it.

Reply:
ok true, good point! I'll do that right now.

Reply:
added in https://github.com/microsoft/LightGBM/pull/2963/commits/3ddcc153ea3596a6cbac475b8f57325e0d53d4cb